acceptance/tests/language/sensitive_data_type.rb

```diff
@@ -75,13 +82,20 @@
           run_assertions(assertion_code, result)
         end
 
-        if agent['platform'] =~ /fedora|centos|el|redhat|scientific|ubuntu|debian|cumulus/
-          step "assert no redacted data in syslog" do
-            #sigh.  gee, thanks for the help, beaker!
-            result = dump_puppet_log(agent)
-            raise if result.length < 3
-            string_io_index = 2
-            run_assertions(assertion_code, result[string_io_index].string)
+        step "assert no redacted data in syslog" do
+          #sigh.  gee, thanks for the help, beaker!
+          syslogfile = case agent['platform']
+                         when /fedora|centos|el|redhat|scientific/ then
+                           '/var/log/messages'
+                         when /ubuntu|debian|cumulus/ then
+                           '/var/log/syslog'
+                         else
+                           nil
+                       end
+          if syslogfile
+            result = agent.exec(Command.new("tail -n 100 #{syslogfile}"),
+                                :acceptable_exit_codes => [0, 1]).stdout.chomp
```

Comment:
why doesn't this just use `on(agent, "tail -n 100 #{syslogfile}").stdout.chomp`
when does this need an exitcode of 1?  empty syslog? syslog < 100 lines? syslogfile doesn't exist?